os/drivers/video/video_null_driver.c

```diff
@@ -0,0 +1,1554 @@
+ *
+ *   This function is called from the video driver using dev ops.
+ *
+ * Input Parameters:
+ *   video_private - Reference to private data.
+ *
+ * Returned Values:
+ *   OK on success, appropriate error code on failures
+ *
+ ****************************************************************************/
+static int video_null_refresh(FAR void *video_private)
+{
+	int ret = OK;
+	return ret;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+/****************************************************************************
+ * Name: video_null_initialize
+ *
+ * Description:
+ *   This completes the initialization operations and register video driver
+ *
+ * Input Parameters:
+ *   devpath -  path to video device
+ *
+ * Returned Values:
+ *   None
+ *
+ ****************************************************************************/
+int video_null_initialize(const char *devpath)
+{
+	int ret = OK;
+
+	struct video_lowerhalf_s *lower = NULL;
+	struct dummy_null_priv_s *priv = NULL;
+
+	/* Allocate the null driver structure */
+	lower = (FAR struct video_lowerhalf_s *)
+			kmm_zalloc(sizeof(struct video_lowerhalf_s));
+	if (!lower) {
+		videovdbg("ERROR: Null driver allocation failed\n");
+		return -ENOMEM;
+	}
+
+	lower->ops = &dummy_null_video_ops;
+
+	/* Allocate the null driver priv data */
+	priv = (FAR struct dummy_null_priv_s *)
+		   kmm_zalloc(sizeof(struct dummy_null_priv_s));
+	if (!lower) {
+		videovdbg("ERROR: Null driver private data allocation failed\n");
+		ret = -ENOMEM;
+		goto errout_with_priv_data;
+	}
+
+	/* Initializing control mapping table */
+	priv->ctrlcount = sizeof(ctrl_mapping) / sizeof(struct v4l2_mapping_s);
+	priv->ctrl = (FAR struct v4l2_mapping_s *)kmm_zalloc(sizeof(ctrl_mapping));
+	if (!priv->ctrl) {
+		videovdbg("ERROR: Null driver control table allocation failed\n");
+		ret = -ENOMEM;
+		goto errout_failure;
+	}
+	memcpy(priv->ctrl, ctrl_mapping, sizeof(ctrl_mapping));
+
+	/* Initialize Format & Frame */
+	priv->fmtcount = sizeof(supported_formats) / sizeof(struct v4l2_format_s);
+	priv->fmt = (FAR struct v4l2_format_s *)kmm_zalloc(sizeof(supported_formats));
+	if (!priv->fmt) {
+		videovdbg("ERROR: Null driver format table allocation failed\n");
+		ret = -ENOMEM;
+		goto errout_failure;
+	}
+	memcpy(priv->fmt, supported_formats, sizeof(supported_formats));
+
+	/* Set current format & frame as the first format & frame */
+	priv->def_format = &priv->fmt[0];
+	priv->cur_format = &priv->fmt[0];
+	priv->cur_frame = &priv->fmt[0].frames[0];
+	/* The initial reference count is 1 */
+	priv->crefs = 1;
+
+	/* Initialize semaphores */
+	sem_init(&priv->sem, 0, 1);
+	lower->dev = priv;
+
+	priv->priv_data = video_register(devpath, (struct video_lowerhalf_s *)lower);
+	if (priv->priv_data == NULL) {
+		videovdbg("Fail to register video driver!!\n");
```

Comment:
You use `videovdbg` with error case. You should use `videodbg`.

Reply:
Done